projects/python-contract-examples/smart_contracts/arc4_types/contract.py

```diff
@@ -0,0 +1,305 @@
+# pyright: reportMissingModuleSource=false
+import typing as t
+
+from algopy import (
+    ARC4Contract,
+    GlobalState,
+    String,
+    UInt64,
+    arc4,
+    urange,
+)
+from algopy.arc4 import abimethod
+
+
+class Arc4Types(ARC4Contract):
+
+    # example: ARC4_UINT64
+    @abimethod()
+    def add_arc4_uint64(self, a: arc4.UInt64, b: arc4.UInt64) -> arc4.UInt64:
+        """
+        Math operations (like a + b) are not supported on arc4.UInt64 types
+        since they are internally represented as byte arrays in the AVM.
+        Use the .native property to perform arithmetic operations.
+        """
+
+        # Use the native integers to perform arithmetic
+        c = a.native + b.native
+
+        return arc4.UInt64(c)
+
+    # example: ARC4_UINT64
+
+    # example: ARC4_UINTN
+    @abimethod()
+    def add_arc4_uint_n(
+        self, a: arc4.UInt8, b: arc4.UInt16, c: arc4.UInt32, d: arc4.UInt64
+    ) -> arc4.UInt64:
+        """
+        The encoding of arc4 integers will be smaller if it uses fewer bits.
+        Ultimately, they are all represented with native UInt64.
+        """
+        assert a.bytes.length == 1
+        assert b.bytes.length == 2
+        assert c.bytes.length == 4
+        assert d.bytes.length == 8
+
+        total = a.native + b.native + c.native + d.native
+
+        return arc4.UInt64(total)
+
+    # example: ARC4_UINTN
+
+    # example: ARC4_BIGUINT
+    @abimethod()
+    def add_arc4_biguint_n(
+        self, a: arc4.UInt128, b: arc4.UInt256, c: arc4.UInt512
+    ) -> arc4.UInt512:
+        """
+        Integers with larger bit size are supported up to 512 bits.
+        Ultimately, they are all represented with native BigUInt.
+        """
+        assert a.bytes.length == 16
+        assert b.bytes.length == 32
+        assert c.bytes.length == 64
+
+        total = a.native + b.native + c.native
+
+        return arc4.UInt512(total)
+
+    # example: ARC4_BIGUINT
+
+    # example: ARC4_BYTES
+    @abimethod()
+    def arc4_byte(self, a: arc4.Byte) -> arc4.Byte:
+        """
+        An arc4.Byte is essentially an alias for an 8-bit integer.
+        """
+        return arc4.Byte(a.native + 1)
+
+    # example: ARC4_BYTES
+
+    # example: ARC4_ADDRESS
+    @abimethod()
+    def arc4_address_properties(self, address: arc4.Address) -> UInt64:
+        underlying_bytes = (  # noqa: F841
```

Comment:
`# noqa: F841` is to ignore unused variables? Maybe log or return `underlying_bytes` and others so you can remove these comments

Reply:
yeah. If I want to do that, I would have to construct a tuple and return various types. I thought that would complicate the example more.